### PR TITLE
fix: redirect uri when session expired

### LINKED ISSUE #2029: Redirect page is not the last view page when session expired and login
## Issue description

When a page is not clicked for a long time and session expired,  browser will be redirected to the login page by XHR :  http://139.217.190.60/user/login?redirect=%2Fuser%2Flogin
the `redirect` param is not the last page user viewed URL and cannot redirect to the old page.

## Environment
APISIX 2.6 with APISIX-dashboard 2.7

### PR DESCRIPTION
**Pull request by foolwc**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
Redirect to the right page when session expired and login success.
**Related issues**
resolve #2029

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
@foolishlwc Please take a look at the CI :)

cc @iamayushdas to take a look also

**Comment by foolwc** [comment]
CI problem fixed. Please take a look and help review @juzhiyuan

File: web/src/app.tsx [hunk 1]
```diff
@@ -33,7 +33,8 @@ export async function getInitialState(): Promise<{
 }> {
   const token = localStorage.getItem('token');
   if (!token) {
-    history.replace(`/user/login?redirect=${encodeURIComponent(window.location.pathname)}`);
+    const redirect = getUrlQuery('redirect');
```
Comment (liuxiran) [inline web/src/app.tsx:36]: image

hi @foolwc , when the url did not have redirect param, we should omit it, and redirect to `/`

Comment (foolwc) [reply web/src/app.tsx]: Got it. Thanks a lot

**Review by bzp2010 — APPROVED** [review]
LGTM

**Review by juzhiyuan — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Comment by liuxiran** [comment]
merged, thanks for your contribution @foolwc

**Pull request merged**